nvflare/private/fed/simulator/simulator_server.py

```diff
@@ -134,3 +134,15 @@ def create_run_manager(self, workspace, job_id):
             client_manager=self.client_manager,
             handlers=self.runner_config.handlers,
         )
+
+    def wait_engine_run_complete(self, name, shutdown_grace_time=2.0, cleanup_grace_time=3.0):
+        while self.engine.engine_info.status != MachineStatus.STOPPED:
+            if self.engine.asked_to_stop:
+                self.engine.engine_info.status = MachineStatus.STOPPED
+
+            time.sleep(3)
```

Comment:
this should use the argument `cleanup_grace_time` ?

Reply:
changed.